### PR TITLE
Add clarification on persistence as suggested by @akhmerov

### PR DESCRIPTION
**Pull request by willingc**
Added clarification based on Gitter discussion.

@akhmerov @minrk

### CODE REVIEW

**Comment by akhmerov** [comment]
Reading the text again I'm actually wondering if it's the correct place for the clarification as opposed to the main [readme](https://github.com/jupyterhub/dockerspawner#dockerspawner-1). After all the image contains nothing about the volume nor about `jupyterhub_config`, so the comments seem a bit out of place here.

**Comment by willingc** [comment]
@akhmerov Reading both readme files. I'm leaning a bit toward rewriting the main readme as the project has evolved over time. I think we can improve the current distinction between dockerspawner and systemuserspawner. Would you mind leaving a few bullets here on why you prefer dockerspawner over systemuserspawner?

cc/ @minrk

**Comment by akhmerov** [comment]
@willingc I'm not a devops, so my opinion is definitely uneducated. Nonetheless, `DockerSpawner` requires less configuration on the host system, and therefore I believe it is easier to migrate/update/cleanly remove. Since you don't create the users on the host system, there is less concern of the host system security, and resource limits can be managed exclusively via docker tools.

One purely cosmetic advantage of `SystemUserSpawner` is due to the docker quirk that maps all internal container users on the host system users. Therefore tools like `top` would show a bunch of processes launched by a single user on the host system. Of course, `SystemUserSpawner` doesn't incur any drawbacks if it is being set up on a server that already has users created (but in that case `SudoSpawner` may be an easier choice).

File: singleuser/README.md [hunk 1]
```diff
@@ -10,6 +10,23 @@ class to launch user notebook servers within docker containers.
...
 This particular server runs (within the container) as the `jupyter` user, with
 home directory at `/home/jupyter`, and the IPython example notebooks at
-`/home/jupyter/examples`. This home directory is *not* persistent by default;
-thus, this image should be used with temporary or demonstration JupyterHub
+`/home/jupyter/examples`.
+
+## Important note on persistence
+
+This home directory, `/home/jupyter`, is *not* persistent; thus,
+this image should be used with temporary or demonstration JupyterHub
```
Comment (minrk) [inline singleuser/README.md]: Maybe something along the lines of "not persistent by default, so some configuration is required unless it is to be used with temporary or demonstration JupyterHub deployments"

**Comment by minrk** [comment]
I think @akhmerov's point about moving more of the details about *how* to do the configuration to the main README is a good idea, and reduce the note on the image README to 'additional configuration is required' if persistence is desirable (rather than the current implication that persistence isn't achievable).

Re: heuristics for SystemUser vs Docker, the gist is: most cases should use Docker. You need a reason to choose SystemUser, and the primary ones are:

1. You are using docker just for environment management, but are running on a system where the users already have accounts and files they should be able to access from within the container.
2. You are using an external service that relies on distinct unix user ownership and permissions (nbgrader)

If neither of those applies, DockerSpawner is probably the right choice.

The simplest version of persistence to the host filesystem is to isolate users in the filesystem, but leave everything owned by the same 'actual' user with DockerSpawner. That is, using docker mounts to isolate user files, not ownership or permissions on the host.

**Comment by willingc** [comment]
@minrk Updated PR with your suggested text. 

I'll work on #123 to incorporate the explanation above.

**Comment by akhmerov** [comment]
Thanks, looks good to me.

**Comment by willingc** [comment]
Thanks for the review @akhmerov on this pr and the other :sunny:

**Pull request merged**